Describe change in discount category match behavior.

mattstein opened the pull request:
This describes the expanded options for applying category-based discounts in Commerce 3, now that [a migration](https://github.com/craftcms/commerce/pull/1565/commits/5bdc7514a564432e0bf40fd4f73b385f8ee76af3) will maintain the previous behavior for discounts that came from Commerce 2.

Commit:
Describe change in discount category match behavior.

Commit:
Reword.

Commit:
Link to Relations + prune cruft.

lukeholder commented:
@mattstein Thinking of relabelling 'Both' to 'Either' in 3.1.11 here: https://github.com/craftcms/commerce/pull/1565

What do you think the option label should be to make the most sense?

mattstein commented:
@lukeholder I think “Either” is probably an improvement since it implies _or_ rather than _and_ (as “Both” might). “Source or Target” would be more explicit, but it also seems a bit clunky.

lukeholder commented:
@mattstein yes we committed to 'Either', and merged it into Commerce 3.1.12.

lukeholder committed:
Upgrade labels

lukeholder commented:
@mattstein just made an update with the labels as pushed into 3.1.12

If you are happy with the above commit I will merge.

Commit:
Shorter.

Avoids duplicating language from control panel, but folds in previous clarification and mention of default.

Commit:
Add missing `you`.

Commit:
+ `and sales`

lukeholder committed:
Merge branch 'v3' into feature/category-matching-upgrade-note

# Conflicts:
#	upgrading.md
